pydantic/_internal/_std_types_schema.py

```diff
@@ -311,35 +164,15 @@ def decimal_prepare_pydantic_annotations(
 
     metadata, remaining_annotations = _known_annotated_metadata.collect_known_metadata(annotations)
 
+    # FIXME: think pydantic-core already checks the config?
     config_allow_inf_nan = config.get('allow_inf_nan')
```

Comment:
I'm not quite familiar enough with how `config` and `metadata` here relate to `pydantic_core`, anyone able to say if this fixme is relevant or I just leave this code here?

Reply:
`config` will be the config for the parent model in this case. metadata is the mapping to unpack into the pydantic-core schema

Reply:
Right, so if I understand correctly we do want to keep this code, and therefore I should just delete the FIXME comment?

Reply:
seems like it to me